django/db/models/functions/text.py

```diff
@@ -188,6 +190,133 @@ def as_sqlite(self, compiler, connection, **extra_context):
         return super().as_sql(compiler, connection, function='UNICODE', **extra_context)
 
 
+class RegexpFlagMixin:
+    def _resolve_flags(self, flags, mapping, default=''):
+        value = default + flags.value
+
+        # Remove duplicate flags preserving the last occurrence of each to
+        # ensures that the last flag is preferred if any are contradictory.
+        value = ''.join(reversed(dict.fromkeys(reversed(value))))
+
+        # Resolve case-sensitivity flags preferring the last flag specified.
+        if {'c', 'i'}.issubset(value):
+            remove = 'i' if value.index('c') > value.index('i') else 'c'
+            value = value.replace(remove, '')
```

Comment:
This is an internal tool, so we can safely assume that we don't have duplicates or conflicted flags. I'd remove it.

Reply:
It isn't internal. The `_resolve_flags()` method is called to smooth out the issues with flags between different backends. Flags can be passed into the different regular expression functions.